lib/Basics/process-utils.cpp

```diff
@@ -225,9 +225,115 @@ static bool CreatePipes(int* pipe_server_to_child, int* pipe_child_to_server) {
 /// @brief starts external process
 ////////////////////////////////////////////////////////////////////////////////
 
-static void StartExternalProcess(ExternalProcess* external, bool usePipes,
-                                 std::vector<std::string> const& additionalEnv,
-                                 std::string const& fileForStdErr) {
+#include <spawn.h>
+
+static void StartExternalProcessPosixSpawn(
+    ExternalProcess* external, bool usePipes,
+    std::vector<std::string> const& additionalEnv,
+    std::string const& fileForStdErr) {
+  int pipe_server_to_child[2];
+  int pipe_child_to_server[2];
+
+  if (usePipes) {
+    bool ok = CreatePipes(pipe_server_to_child, pipe_child_to_server);
+
+    if (!ok) {
+      external->_status = TRI_EXT_PIPE_FAILED;
+      return;
+    }
+  }
+
+  posix_spawn_file_actions_t file_actions;
+  posix_spawn_file_actions_init(&file_actions);
+  // file actions are performed in order they were added.
+
+  if (usePipes) {
+    posix_spawn_file_actions_adddup2(&file_actions, pipe_server_to_child[0], 0);
+    posix_spawn_file_actions_adddup2(&file_actions, pipe_child_to_server[1], 1);
+
+    posix_spawn_file_actions_addclose(&file_actions, pipe_server_to_child[0]);
+    posix_spawn_file_actions_addclose(&file_actions, pipe_server_to_child[1]);
+    posix_spawn_file_actions_addclose(&file_actions, pipe_child_to_server[0]);
+    posix_spawn_file_actions_addclose(&file_actions, pipe_child_to_server[1]);
```

Comment:
Note: all of these function calls can fail, which is indicated by any of the calls returning a value other than zero. Do we want to add error handling here? I guess in case of an error all we can do set `external->_status = TRI_EXT_PIPE_FAILED;`, log a message and error out.

Reply:
I have added error handling.